### PR TITLE
Model local restriction data

### PR DESCRIPTION
**Pull request by JonathanHallam**
Adds a service that takes a GSS code (from Mapit), and matches against the local lockdown data held in finder-frontend, and returns data for that area.

We have chosen to stub out the fixture path in order to separate test data from that which is live.

## Testing locally

The test suite can be run with `bundle exec rake`. You can also open a rails console and check that the LocalRestriction class is responding to the expected methods. 

```
irb(main):004:0> restriction = LocalRestriction.new("E08000001")
=> #<LocalRestriction:0x00007fc98d9b7c20 @gss_code="E08000001">
irb(main):005:0> restriction.area_name
=> "Bolton"
```

## Trello 
https://trello.com/c/40EHq00z/829-add-mapping-service-to-get-lockdown-data-from-gss-code

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/finder-frontend), after merging.

### CODE REVIEW

File: app/models/local_restriction.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    area_data["end_date"].to_date
+  end
+
+private
+
+  def restriction_data
+    @restriction_data ||= YAML.load_file(FILE_PATH)
+  end
+
+  def area_data
```
Comment (leenagupte) [inline app/models/local_restriction.rb:40]: I wonder if we might need to make this public later, so that we can check if any restrictions exist for a gss code. i.e. `LocalRestriction.new("E000000001").area_data.present?` 🤔

File: spec/fixtures/local-restrictions.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+E08000001:
+  alert_level: 4
+  name: Tatooine
```
Comment (leenagupte) [inline spec/fixtures/local-restrictions.yaml:4]: ❤️

File: app/models/local_restriction.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    area_data["start_date"].to_date
+  end
+
+  def end_date
+    area_data["end_date"].to_date
+  end
+
+private
+
+  def restriction_data
```
Comment (leenagupte) [inline app/models/local_restriction.rb:36]:
```suggestion
  def all_restrictions
```

File: app/models/local_restriction.rb (hunk 1 above)
Comment (leenagupte) [inline app/models/local_restriction.rb:40]:
```suggestion
  def restriction
```

File: spec/models/local_restriction_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+require "spec_helper"
+
+RSpec.describe LocalRestriction do
+  before do
+    stub_const("LocalRestriction::FILE_PATH", "spec/fixtures/local-restrictions.yaml")
```
Comment (leenagupte) [inline spec/models/local_restriction_spec.rb:5]: Nice!

**Review by leenagupte — COMMENTED** [review]
Looking good!
A couple of naming thoughts, but nothing major.

File: app/models/local_restriction.rb (hunk 1 above)
Comment (JonathanHallam) [reply app/models/local_restriction.rb:40]: Yeah, good point. I guess we could use a proxy like `LocalRestriction.new("E000000001").area_name.present?` but it's not as clear.

File: app/models/local_restriction.rb (hunk 3 above)
Comment (JonathanHallam) [reply app/models/local_restriction.rb]: Good shout 👍

File: app/models/local_restriction.rb (hunk 1 above)
Comment (JonathanHallam) [reply app/models/local_restriction.rb]: done! 🥇

File: app/models/local_restriction.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  def guidance
+    restriction["guidance"]
+  end
+
+  def extra_restrictions
+    restriction["extra_restrictions"]
+  end
+
+  def start_date
+    restriction["start_date"].to_date
```
Comment (leenagupte) [inline app/models/local_restriction.rb:27]: I think we need to add tests for when start_date and end_date is nil:

```
irb(main):005:0> rst = LocalRestriction.new("E06000008")
=> #<LocalRestriction:0x0000562becd45eb8 @gss_code="E06000008">
irb(main):006:0> rst.start_date
Traceback (most recent call last):
        2: from (irb):6
        1: from app/models/local_restriction.rb:27:in `start_date'
NoMethodError (undefined method `to_date' for nil:NilClass)
Did you mean?  to_d
```

It should be possible to fix this with a safe navigation operator, e.g.:

```
restriction["start_date"]&.to_date
```

**Review by leenagupte — CHANGES_REQUESTED** [review]
`start_date` and `end_date` throw an error if they're nil. Once that's fixed this is good to go!

File: app/models/local_restriction.rb (hunk 5 above)
Comment (JonathanHallam) [reply app/models/local_restriction.rb:27]: ooh good spot, sorry! Been fixed now and tests added :)

**Review by leenagupte — APPROVED** [review]
